File: dotnet/source/Templates/WebAPI/Devon4Net.WebAPI.Implementation/Business/VisitorManagement/ETO/VisitorETO.cs
using Devon4Net.WebAPI.Implementation.Business.AccessCodeManagement.ETO;
using System;
using System.Collections.Generic;
using Newtonsoft.Json;

namespace Devon4Net.WebAPI.Implementation.Business.VisitorManagement.ETO
{
    public partial class VisitorETO
    {
        public long id { get; set; }
        public int modificationCounter { get; set; }
        public string username { get; set; }
        public string name { get; set; }
        public string password { get; set; }
        public string phoneNumber { get; set; }
        public bool? acceptedCommercial { get; set; }
        public bool acceptedTerms { get; set; }
        public bool? userType { get; set; }
    }
}

File: push_to_the_edges/edge_server/server/src/core/visitormanagement/logic/api/visitor_eto.rs
use serde::{Serialize, Deserialize};
use validator::Validate;
use crate::api::common::logic::api::eto::EntityETO;
use crate::api::common::rest::api::saveable::Saveable;

#[derive(Deserialize, Serialize, Validate, Clone)]
#[serde(rename_all="camelCase")]
pub struct VisitorEto {
    pub id: Option<i64>,
    pub modification_counter: Option<i32>,
    #[validate(email(code="mail", message="Email format not valid"))]
    pub username: Option<String>,
    pub name: Option<String>,
    pub phone_number: Option<String>,
    pub password: Option<String>,
    pub accepted_commercial: Option<bool>,
    pub accepted_terms: bool,
    pub user_type: Option<bool>,
}

impl EntityETO for VisitorEto {}
impl Saveable for VisitorEto {}